planner/core/task.go

```diff
@@ -1820,6 +1828,10 @@ func (p *PhysicalHashAgg) attach2TaskForMpp(tasks ...task) task {
 	}
 	switch p.MppRunMode {
 	case Mpp1Phase:
+		// TiFlash will check if agg func mode are all same. Otherwise will give error.
+		if !aggFuncModeSame(p) {
```

Comment:
In which case that the agg func mode in the same aggregation will be different?

Reply:
In [rule_aggregatoin_pushdown.go](https://github.com/pingcap/tidb/blob/master/planner/core/rule_aggregation_push_down.go#L404), will add `firstrow(CompleteMode)`, but other agg funcs are PartionMode.

Reply:
Make `firstrow` as `Partial1Mode` [here](https://github.com/pingcap/tidb/pull/40828/files#diff-e404d5c17cb93070fc630bc941121e103d5739b70b62e06df93d6a8861315684R408)